guides/runtime-parser.md

```diff
@@ -0,0 +1,308 @@
+      {{title}}
+    </md-toolbar>
+    <md-spinner *shellRender></md-spinner>
+    <h1 *shellNoRender>App is Fully Rendered</h1>
+  `,
+  styles: [`
+    md-spinner {
+      margin: 24px auto 0;
+    }
+    .icon {
+      width: 40px;
+      height: 40px;
+      display: inline-block;
+    }
+    .icon.ng {
+      background-image: url(./images/angular.png);
+    }
+  `],
+  directives: [APP_SHELL_DIRECTIVES, MdToolbar, MdSpinner]
+})
+export class HelloMobileAppComponent {
+  title = 'Hello Mobile';
+}
+```
+
+...and in Universal, we bootstrap our application in the following way:
+
+```typescript
+import { provide } from '@angular/core';
+import { APP_BASE_HREF } from '@angular/common';
+import { APP_SHELL_RUNTIME_PROVIDERS } from '@angular/app-shell';
+import { HelloMobileAppComponent } from './app';
+import {
+  REQUEST_URL,
+  ORIGIN_URL
+} from 'angular2-universal';
+
+export const options = {
+  directives: [
+    // The component that will become the main App Shell
+    HelloMobileAppComponent
+  ],
+  platformProviders: [
+    APP_SHELL_RUNTIME_PROVIDERS,
+    provide(ORIGIN_URL, {
+      useValue: ''
+    })
+  ],
+  providers: [
+    // What URL should Angular be treating the app as if navigating
+    provide(APP_BASE_HREF, {useValue: '/'}),
+    provide(REQUEST_URL, {useValue: '/'})
+  ],
+  async: false,
+  preboot: false
+};
+```
+
+In the example above, we prerender the component using Universal and the `APP_SHELL_RUNTIME_PROVIDERS`. Once the user opens our prerendered application she will see the following user interface:
+
+<img src="img/prerendered-universal.png" width="300" alt="Prerendered page by Universal">
+
+This is how the app will look once it has been completely rendered as well. However, for our App Shell we want to provide only the minimal UI which shows that the app is working and its initialization is in progress. One way to do this is by using the `APP_SHELL_BUILD_PROVIDERS` instead of `APP_SHELL_RUNTIME_PROVIDERS`. This way Universal will strip the content marked with the `shellNoRender` directive and output only the part of the application that is intended to be visualized as App Shell.
+
+Unfortunately, this way we introduce the following problems:
+
+- We cannot reuse the given view as the App Shell for other routes in our application.
+- We must annotate the template of each individual route component which should be progressively loaded with `shellRender` and `shellNoRender` directives.
+
+On top of that, if the final App Shell has references to external images, users with slow Internet connection will not have the best experience possible. For instance, in the example above the Angular logo in the header is an external resource which needs to be fetched from the network.
+
+By using the Angular App Shell Runtime Parser in a Service Worker we can solve all of these issues! Lets see how!
+
+# Exploring the Runtime Parser
+
+First, lets take a look at a sample Service Worker which uses the Runtime Parser:
```

Comment:
Since these guides are in the context of the mobile toolkit, and Angular CLI, all examples should be based on what's already been done by CLI. I.e. there is already a worker installed that handles file caching and what-not, how now can I use the parser with that? Do I need to provide a new worker that imports other workers? Same goes for setting up Universal, more hands-on instructions with all necessary steps and information to reproduce.